libcontainer/container_linux.go

```diff
@@ -2246,6 +2250,34 @@ func (c *Container) bootstrapData(cloneFlags uintptr, nsMaps map[configs.Namespa
 		})
 	}
 
+	// write boottime and monotonic offsets
+	// 0 is default value in /proc/PID/timens_offsets
+	// if 0, do not bootstrap data
+	if c.config.TimeOffsets != nil {
+
+		if c.config.TimeOffsets["boottime"].Nanosecs != 0 || c.config.TimeOffsets["boottime"].Secs != 0 {
+			b, err := encodeTimeNs(c.config.TimeOffsets["boottime"])
```

Comment:
You use 3 map lookups here, while 1 should be enough. Make encodeTimeNs check if the fields are all 0 and return empty string in such case, add a check for empty string below.

Reply:
It also looks like this would be a good place to write something like
```go
for k, v := c.Config.TimeOffsets {
...
}
```

making the code more generic and avoiding to use specific strings.